### PR TITLE
[transform] Add tensor padding option in tensor_transform

### PR DESCRIPTION
**Pull request by niley7464**
This PR adds tensor padding mode in tensor_transform.
tensor padding has options like this.
left, right, top, bottom, front, back, type (NCHW or NHWC).

See also: #4106 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/elements/gsttensor_transform.h [hunk 1]
```diff
@@ -83,6 +84,17 @@ typedef enum
...
+typedef enum
+{
+  PADDING_LEFT = 0,
+  PADDING_RIGHT = 1,
+  PADDING_TOP = 2,
+  PADDING_BOTTOM = 3,
+  PADDING_FRONT = 4,
+  PADDING_BACK = 5,
+  PADDING_END,
+} tensor_transform_padding_axis;
```
Comment (myungjoo) [inline gst/nnstreamer/elements/gsttensor_transform.h:96]: How about...
```
  PADDING_LEFT = 0x1,
  PADDING_RIGHT = 0x2,
  PADDING_TOP = 0x4,
  PADDING_BOTTOM = 0x8,
  PADDING_AROUND_PLAIN = 0xF
  PADDING_FRONT = 0x10,
  PADDING_BACK = 0x20,
  PADDING_WRAP_CUBE = 0x3F
```
?

**Review by myungjoo — COMMENTED** [review]
1. You do not know which dimension is height, width, and depth. How are you going to identify whether the given tensor is NCHW or NHWC or ...?
2. We may also have a syntax such as : "dim0:-1:+2.dim1:-3:+4" to add 1 to "left", 2 to "right", 3 to "top", and 4 to "bottom" if dim0 is W and dim1 is H.

**Comment by niley7464** [comment]
> You do not know which dimension is height, width, and depth. How are you going to identify whether the given tensor is NCHW or NHWC or ...?

I will add layout in option. Thank you

> We may also have a syntax such as : "dim0:-1:+2.dim1:-3:+4" to add 1 to "left", 2 to "right", 3 to "top", and 4 to "bottom" if dim0 is W and dim1 is H.

Right. jaeyun said something similar, so I am planning to change the options to include names rather than just arrays.

File: gst/nnstreamer/elements/gsttensor_transform.c [hunk 2]
```diff
@@ -73,6 +73,7 @@ GST_DEBUG_CATEGORY_STATIC (gst_tensor_transform_debug);
 #define REGEX_STAND_OPTION "^(default|dc-average)(:([u]?int(8|16|32|64)|float(16|32|64)))?(,per-channel:(true|false))?$"
 #define REGEX_CLAMP_OPTION "^((([-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?))):"\
     "((([-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?)))$"
+#define REGEX_PADDING_OPTION "^((left|right|top|bottom|front|back):(\\d)(,)?)+(type:(NCHW|NHWC))?$"
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_transform.c:76]: suggestion: type > layout

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]

**Comment by myungjoo** [comment]
Now, you may proceed with ORC! :)

**Pull request merged**